homeassistant/components/tts/__init__.py

```diff
@@ -345,6 +342,19 @@ async def async_clear_cache_handle(service: ServiceCall) -> None:
         schema=SCHEMA_SERVICE_CLEAR_CACHE,
     )
 
+    platform_setups = await async_setup_legacy(hass, config)
+
+    # We need to add the component here break the deadlock
+    # when setting up integrations from config entries as
+    # they would otherwise wait for the tts to be
+    # setup and thus the config entries would not be able to
+    # setup their platforms.
+    hass.config.components.add(DOMAIN)
```

Comment:
This was a bug that tts was missing this (`device_tracker` and `notify` have the same pattern)